drivers/hmis/app/models/hmis/form/definition.rb

```diff
@@ -46,6 +46,9 @@ class Hmis::Form::Definition < ::GrdaWarehouseBase
   has_many :external_form_publications, class_name: 'HmisExternalApis::ExternalForms::FormPublication', dependent: :destroy
   has_many :custom_data_element_definitions, class_name: 'Hmis::Hud::CustomDataElementDefinition', dependent: :nullify,
                                              primary_key: 'identifier', foreign_key: 'form_definition_identifier'
+  has_one :published_version, -> { published }, class_name: 'Hmis::Form::Definition', primary_key: 'identifier', foreign_key: 'identifier'
+  has_one :draft_version, -> { draft }, class_name: 'Hmis::Form::Definition', primary_key: 'identifier', foreign_key: 'identifier'
```

Comment:
It looks like if there were multiple draft/published versions you might get an inconsistent record. Perhaps we should have a constraint to enforce that, otherwise adding an order should fix it
```suggestion
  has_one :published_version, -> { order(version: :desc).published }, class_name: 'Hmis::Form::Definition', primary_key: 'identifier', foreign_key: 'identifier'
  has_one :draft_version, -> { order(version: :desc).draft }, class_name: 'Hmis::Form::Definition', primary_key: 'identifier', foreign_key: 'identifier'
```

Reply:
I think we plan to have a uniqueness constraint - @gigxz added a note to do that on this ticket https://github.com/open-path/Green-River/issues/6006

However, I'm happy to add the order here too just to be sure.

Reply:
I think we plan to have a uniqueness constraint - @gigxz added a note to do that on this ticket https://github.com/open-path/Green-River/issues/6006

However, I'm happy to add the order here too just to be sure.